File: bench/dict_bench.lua
local start = os.clock()
local h = {}
local i = 0
while i < 200000 do
    h["key_" .. i] = i
    i = i + 1
end
local sum = 0
i = 0
while i < 200000 do
    sum = sum + h["key_" .. i]
    i = i + 1
end
local dur = os.clock() - start
print(string.format("dict 200k insert+lookup: %fs, sum=%d", dur, sum))

File: bench/dict_bench.py
import time

start = time.perf_counter()
h = {}
i = 0
while i < 800000:
    h[f"key_{i}"] = i
    i += 1
total = 0
i = 0
while i < 800000:
    total += h[f"key_{i}"]
    i += 1
dur = time.perf_counter() - start
print(f"dict 200k insert+lookup: {dur}s, sum={total}")
